sdk/nodejs/resource.ts

```diff
@@ -89,6 +104,10 @@ export abstract class Resource {
                 throw new RunError(`Resource parent is not a valid Resource: ${opts.parent}`);
             }
 
+            this.__parentResource = opts.parent;
+            this.__parentResource.__childResources = this.__parentResource.__childResources || new Set();
+            this.__parentResource.__childResources.add(this);
```

Comment:
So this makes sense and is great, but since it's my problem I have to ask: I'm really afraid of doing stuff like this when, in the near term, it's very possible that the parent and child resources could be created in two separate processes, in two separate languages. It seems like we'd basically be screwed if this were the case:

1. "Remote" children of resources would not be able to register themselves with the parent, since that involves mutating an object in another process,
2. Parents who create "remote" resources won't be aware of them, since it's the child's responsibility to add itself to the list of child resources in the parent

In my mind this imposes a heavy constraint on a multi-language interop story: we actually need to remote resource objects and expose them as "handles", and both the caller and the callee language host need to be able to call into each other at-will. Furthermore, we'll have to be absolutely sure that every language implements this the exact same way, since it would be possible for children to remote into a parent implementation in a different language while still expecting the host language's semantics.

I love the idea here, it makes sense to me, but I do wonder if it's possible to store this state in a different location. Perhaps in the engine? Some way such that we have a well-defined, language-independent protocol of accessing the set of resources that are parented to a particular resource?

This sort-of depends on our strategy for multi-language interop, so I get that it's hard to worry about hypothetical issues for a system that doesn't yet exist, but I worry that this is going to work great for Node and be a massive headache as soon as we start moving into a multi-process world.